src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -840,6 +840,15 @@ fun getPartitioningSelector(
         idColumn: String
 ) = "partitions[ 1 + ((array_length(partitions,1) + (('x'||right(${idColumn}::text,8))::bit(32)::int % array_length(partitions,1))) % array_length(partitions,1))]"
 
+/**
+ * 1. partitions
+ * 2. array length ( partitions )
+ * 3. array length ( partitions )
+ * 4. array length ( partitions )
+ */
+fun getDirectPartitioningSelector(
+        idColumn: String
+) = "(?)[ 1 + ((? + (('x'||right(${idColumn}::text,8))::bit(32)::int % ?)) % ?)]"
```

Comment:
It appears as though this pattern has been here but can you add a line for what its doing in the javadoc? I cannot decipher it 😆

Reply:
Ha! Sure. I barely remember it myself. It converts uuid to text, takes the right-most 8 characters, prepends an x, casts it to 32 bit string interpreting it as hex, cast those 32 bits as an int, then uses the size of partition to compute the positive mod index into the array.

Reply:
👍